### PR TITLE
[hap2] Create new nagios plugin that use livestatus API

### PR DESCRIPTION
**Pull request by Mnakagawa**
livestatus API:
https://mathias-kettner.de/checkmk_livestatus.html

CentOS(from epel repository)
```
# yum install check-mk-livestatus
```

Ubuntu
```
# apt-get install check-mk-livestatus
```

Python library:
https://github.com/arthru/python-mk-livestatus
CentOS & Ubuntu
```
# pip install python-mk-livestatus
```

### CODE REVIEW

**Comment by Mnakagawa** [comment]
Some test are failed, but this failure is not coused by hap2_nagios_livestatus.
OTOH, some test are sucess. I think, we can merge this PR.
What do you think?

File: server/hap2/hatohol/hap2_nagios_livestatus.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+from hatohol import hap
+from hatohol import haplib
+from hatohol.haplib import Utils
+from hatohol import standardhap
+import socket
+import uuid
+import json
+from mk_livestatus import Socket
+
+logger = getLogger(__name__)
```
Comment (cosmo0920) [inline server/hap2/hatohol/hap2_nagios_livestatus.py]: This line contains a same problem of #1876.

File: server/data/server-type-hap2-nagios-livestatus.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+  ON DUPLICATE KEY UPDATE name='Nagios Status (HAPI2)', parameters='[{"id": "nickname", "label": "Nickname"}, {"hint": "IP_ADDRESS:PORT or SOCKET_DIRECTORY", "id": "baseURL", "label": "URL"}, {"id": "userName", "label": "User name"}, {"inputStyle": "password", "id": "password", "label": "Password"}, {"default": "30", "id": "pollingInterval", "label": "Polling interval (sec)"}, {"default": "10", "id": "retryInterval", "label": "Retry interval (sec)"}, {"inputStyle": "checkBox", "id": "passiveMode", "label": "Passive mode"}, {"hint": "amqp://user:password@localhost[:port]/vhost", "id": "brokerUrl", "label": "Broker URL"}, {"hint": "(empty: Default)", "allowEmpty": true, "id": "staticQueueAddress", "label": "Static queue address"}, {"allowEmpty": true, "id": "tlsCertificatePath", "label": "TLS client certificate path"}, {"allowEmpty": true, "id": "tlsKeyPath", "label": "TLS client key path"}, {"allowEmpty": true, "id": "tlsCACertificatePath", "label": "TLS CA certificate path"}, {"inputStyle": "checkBox", "allowEmpty": true, "id": "tlsEnableVerify", "label": "TLS: Enable verify"}]', plugin_path='start-stop-hap2-nagios-livestatus.sh', plugin_sql_version='1', plugin_enabled='1';
```
Comment (cosmo0920) [inline server/data/server-type-hap2-nagios-livestatus.sql]: Is 'Nagios Status (HAPI2)' typo of 'Nagios LiveStatus (HAPI2)' ?

File: server/data/server-type-hap2-nagios-livestatus.sql [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+  VALUES (7, 'Nagios Livestatus (HAPI2)', '[{"id": "nickname", "label": "Nickname"}, {"hint": "IP_ADDRESS:PORT or SOCKET_DIRECTORY", "id": "baseURL", "label": "URL"}, {"id": "userName", "label": "User name"}, {"inputStyle": "password", "id": "password", "label": "Password"}, {"default": "30", "id": "pollingInterval", "label": "Polling interval (sec)"}, {"default": "10", "id": "retryInterval", "label": "Retry interval (sec)"}, {"inputStyle": "checkBox", "id": "passiveMode", "label": "Passive mode"}, {"hint": "amqp://user:password@localhost/vhost", "id": "brokerUrl", "label": "Broker URL"}, {"hint": "(empty: Default)", "allowEmpty": true, "id": "staticQueueAddress", "label": "Static queue address"}, {"allowEmpty": true, "id": "tlsCertificatePath", "label": "TLS client certificate path"}, {"allowEmpty": true, "id": "tlsKeyPath", "label": "TLS client key path"}, {"allowEmpty": true, "id": "tlsCACertificatePath", "label": "TLS CA certificate path"}, {"inputStyle": "checkBox", "allowEmpty": true, "id": "tlsEnableVerify", "label": "TLS: Enable verify"}]', 'start-stop-hap2-nagios-livestatus.sh', 1, 1, '6f024e3e-a2cd-11e5-bfc7-d43d7e3146fb')
```
Comment (cosmo0920) [inline server/data/server-type-hap2-nagios-livestatus.sql]: This plugin name should be unified with 'Nagios LiveStatus (HAPI2)' like this line: https://github.com/project-hatohol/hatohol/pull/1869/files#diff-1d36e067eee9509569458ed441fe9e23R2

File: server/hap2/hatohol/hap2_nagios_livestatus.py [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+    def create_main_plugin(self, *args, **kwargs):
+        return Hap2NagiosLivestatusMain(*args, **kwargs)
+
+    def create_poller(self, *args, **kwargs):
+        return Hap2NagiosLivestatusPoller(self, *args, **kwargs)
+
+
+if __name__ == '__main__':
+    hap = Hap2NagiosLivestatus()
+    hap()
```
Comment (cosmo0920) [inline server/hap2/hatohol/hap2_nagios_livestatus.py]: `hap` has a possibility to be collided for global defined `hap`. see: https://github.com/project-hatohol/hatohol/issues/1877

**Comment by Mnakagawa** [comment]
@cosmo0920 Thank you for your comment. I fixed to pointed out the place.
I send commit after merge #1882.

**Comment by cosmo0920** [comment]
#1822 is already merged into master.
@Mnakagawa -san, please rebase with current master and push your fix commits.

**Comment by Mnakagawa** [comment]
Sure.
I create new PR

**Pull request closed without merging**